test/e2e/scalability_test.go

```diff
@@ -0,0 +1,57 @@
+package e2e
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	bmov1alpha1 "github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+/*
+ * Apply scaled number of BMHs per batches and wait for the them to become available
+ * The test pass when all the BMHs become available
+ */
+
+var _ = Describe("When testing scalability with fakeIPA [scalability]", Label("scalability"), func() {
+	scalabilityTest()
+})
+
+func scalabilityTest() {
+	It("Should apply per batches BMHs wait for them to become available", func() {
+		numNodes, _ := strconv.Atoi(e2eConfig.GetVariable("NUM_NODES"))
+		batch, _ := strconv.Atoi(e2eConfig.GetVariable("BMH_BATCH_SIZE"))
+		Logf("Starting scalability test")
+		bootstrapClient := bootstrapClusterProxy.GetClient()
+		ListBareMetalHosts(ctx, bootstrapClient, client.InNamespace(namespace))
+		applyBatchBmh := func(from int, to int) {
+			Logf("Apply BMH batch from node_%d to node_%d", from, to)
```

Comment:
what is the reason that we are applying BMHs in batches? We have a `PROVISIONING_LIMIT` in [BMO](https://github.com/metal3-io/baremetal-operator/blob/main/docs/configuration.md?plain=1#L38). Can we utilize that?

Reply:
I think it may be a good idea to have batches still. It gives us "checkpoints" that could be helpful when determining if we have a scalability issue or just some other random issue. E.g. if the first 60 BMHs were ok but after that it failed, then it could be a scalability issue. But if it fails directly for the first 20 then it is likely some other issue.

Reply:
That is true, I am ok with this. Since BMO already has a safety net there and I am trusting that https://github.com/metal3-io/metal3-docs/blob/main/design/baremetal-operator/limit-hosts-provisioning.md added the limit based on some statistics , I thought batching here wont actually add much value.

Reply:
ok if @kashifest agree we can merge it

Reply:
/approve